### ISSUE TITLE
Hyperparameter optimisation

### ISSUE THREAD

**Issue by heyzx123** [ultralytics/yolov5#10145]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi，I have a question about hyperparameter optimisation.For hyperparametric optimization of the model, how many epoch training sessions are required for an individual to obtain the evaluation metric as its fitness value?

**Comment by glenn-jocher** [comment]
@heyzx123 fitness is defined for every epoch as a weighted combination of metrics (mostly mAP50-95). 

Evolution is agnostic to your base scenario. You can evolve any base scenario, i.e. 1 epoch or a million epochs. The hyps that are evolved will be optimized for that specific base scenario.

**Comment by heyzx123** [comment]
Meaning I can optimize with 10 epochs, or 100 epochs of results (mAP50-95) as fitness? Does it depend on the number of epochs needed for the main optimization task?

**Comment by glenn-jocher** [comment]
@heyzx123 if you want to optimize a model to do X, then you do python train.py X --evolve.

X is entirely up to you, I can't offer any input there.

**Comment by heyzx123** [comment]
ok,thank you!

**Comment by glenn-jocher** [comment]
@heyzx123 you're welcome! If you have any other questions, feel free to ask. Good luck with your hyperparameter optimization!
